mava/wrappers/jumanji.py

```diff
@@ -169,28 +136,34 @@ def observation_spec(self) -> specs.Spec[Observation]:
         )
 
 
-class RwareMultiAgentWrapper(Wrapper):
+class RwareWrapper(Wrapper):
     """Multi-agent wrapper for the Robotic Warehouse environment."""
 
-    def reset(self, key: chex.PRNGKey) -> Tuple[State, TimeStep]:
-        """Reset the environment. Updates the step count."""
-        state, timestep = self._env.reset(key)
-        timestep.observation = Observation(
+    def __init__(self, env: RobotWarehouse):
+        super().__init__(env)
+        self._env: RobotWarehouse
+
+    def modify_timestep(self, timestep: TimeStep) -> TimeStep[Observation]:
+        n_agents = self._env.num_agents
+        observation = Observation(
             agents_view=timestep.observation.agents_view,
             action_mask=timestep.observation.action_mask,
-            step_count=jnp.repeat(timestep.observation.step_count, self._env.num_agents),
+            step_count=jnp.repeat(timestep.observation.step_count, n_agents),
         )
-        return state, timestep
+        shared_reward = jnp.sum(timestep.reward)
```

Comment:
Do we need this? I think separate rewards for each agent is a good thing allows them to learn independently

Reply:
I think we need to add it as a feature in the systems files instead of the wrapper, what do you think?

Reply:
Oh it's a mistake I added that to test the LBF environment I'll remove it and add this in the LBF wrapper :+1:.
For the idea of using separate rewards or the shared in LBF, we can use a flag as Omayma mentioned that will be helpful in future to test networks with non-shared parameters :crossed_fingers: